pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -552,7 +552,10 @@ func getLocalIP() ([]v1.NodeAddress, error) {
 			klog.Warningf("Failed to extract addresses for NodeAddresses - %v", err)
 		} else {
 			for _, addr := range localAddrs {
-				if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+				if i.Flags&net.FlagLoopback != 0 {
```

Comment:
Can we move this check before fetching addresses from interface?

Reply:
Just updated it to do that